interactive.c

```diff
@@ -173,23 +225,69 @@ void interactiveShowData(void) {
             }
 
             if (trackDataValid(&a->airground_valid) && a->airground == AG_GROUND) {
-                snprintf(strFl, 7," grnd");
+                snprintf(strFl, 7,"grnd ");
             } else if (Modes.use_gnss && trackDataValid(&a->altitude_geom_valid)) {
                 snprintf(strFl, 7, "%5dH", convert_altitude(a->altitude_geom));
             } else if (trackDataValid(&a->altitude_baro_valid)) {
                 snprintf(strFl, 7, "%5d ", convert_altitude(a->altitude_baro));
             }
+            double signalDisplay = 10.0 * log10(signalAverage);
+
+            double distance = 0.0;
+            char strDistance[8]           = " ";
+            double bearing = 0.0;
+            char strBearing[9]           = " ";
+
+            if ((Modes.bUserFlags & MODES_USER_LATLON_VALID) && trackDataValid(&a->position_valid)) {
+                distance = greatcircle(Modes.fUserLat, Modes.fUserLon,
+                    a->lat, a->lon);
+
+                distance /= 1000.0;
+
+                distance *= distance_units_conversion;
+                if (distance > distanceMax) {
+                    distanceMax = distance;
+                }
+                snprintf(strDistance, 8, "%5.1f ", distance);
+                bearing = get_bearing(Modes.fUserLat, Modes.fUserLon, a->lat, a->lon);
+                snprintf(strBearing, 9, "%5.0f ", bearing);
```

Comment:
Let's use `sizeof(strWhatever)` not a magic constant here (I know, the existing code is not good about doing that)

Reply:
Yeah, it's hard to know what the policy is for a project you're new to so I  just default to copying the existing patterns.  :)
I'll fix.